apps/charterafrica/src/lib/ecosystem/ecosystem.js

```diff
@@ -0,0 +1,104 @@
+  TOOL_COLLECTION,
+  ECOSYSTEM_GLOBAL,
+} from "@/charterafrica/lib/ecosystem/models";
+
+const bulkMarkDeleted = async (collection, fromSource) => {
+  const dataIds = fromSource.map(({ id }) => id).join(",");
+  const { docs: toDelete } = await api.getCollection(collection, {
+    where: {
+      airtableId: {
+        not_in: dataIds,
+      },
+    },
+  });
+  Promise.all(
+    toDelete.map(async ({ id }) => {
+      await api.updateCollection(collection, id, {
+        deletedAt: new Date(),
+      });
+    })
+  );
+};
+const processTools = async (config, tableData) => {
+  const { tools: toolsFromAirtable } = tableData;
+  await bulkMarkDeleted(TOOL_COLLECTION, toolsFromAirtable);
+  const processedToolPromises = toolsFromAirtable.map(async (data) => {
+    const airtableData = await processToolFromAirtable(
+      {
+        ...data.fields,
+        id: data.id,
+      },
+      config,
+      tableData
+    );
+    // Only get git data if not exist in database
+    return createCollection(TOOL_COLLECTION, airtableData, config);
+  });
+  return Promise.allSettled(processedToolPromises);
+};
+
+const processOrganisations = async (config, tableData) => {
+  const { organisations: organisationsFromAirtable } = tableData;
+  await bulkMarkDeleted(ORGANIZATION_COLLECTION, organisationsFromAirtable);
+  const processedOrgPromises = organisationsFromAirtable.map(async (data) => {
+    const airtableData = await processOrganisationFromAirTable(
+      {
+        ...data.fields,
+        id: data.id,
+      },
+      config,
+      tableData
+    );
+    // Only get git data if not exist in database
+    return createCollection(ORGANIZATION_COLLECTION, airtableData, config);
+  });
+  return Promise.allSettled(processedOrgPromises);
+};
+
+const processContributors = async (config, tableData) => {
+  const { contributors: contributorsFromAirtTable } = tableData;
+  await bulkMarkDeleted(CONTRIBUTORS_COLLECTION, contributorsFromAirtTable);
+  const processedContributors = contributorsFromAirtTable.map(async (data) => {
+    const airtableData = await processContributorFromAirtable(
+      {
+        ...data.fields,
+        id: data.id,
+      },
+      config,
+      tableData
+    );
+    // Only get git data if not exist in database
+    return createCollection(CONTRIBUTORS_COLLECTION, airtableData, config);
+  });
+  return Promise.allSettled(processedContributors);
+};
+
+export const updateEcosystemContent = async (req, res) => {
+  // For all list in database query Github API. using ETAG
+  res.status(200).json({});
+};
+
+const execute = async () => {
+  const config = await api.findGlobal(ECOSYSTEM_GLOBAL, {});
+  const tableData = await getAirtableData(config);
+  await processContributors(config, tableData);
+  await processTools(config, tableData);
+  await processOrganisations(config, tableData);
+};
+
+export const updateEcosystemList = async () => {
+  execute();
+  return { message: "PROCESS_STARTED" };
```

Comment:
Any way of knowing when the process finishes?

Reply:
No, I get that & I agree @koechkevin. What I'm asking is, as an SRE, is there any way of knowing that a this process has completed?

It's not a high priority since it doesn't affect end users but it's important for us to be able to measure & tell how long it takes.

Reply:
@kilemensi  I have updated this with `Sentry.captureEvent`

Reply:
I guess it's cool for a start but I was talking about [performance monitoring](https://docs.sentry.io/platforms/javascript/guides/nextjs/performance/) @koechkevin